qa/L0_memory_growth/busy_op_test.py

```diff
@@ -71,47 +62,40 @@
         required=True,
         help="Number of asynchronous requests to launch.",
     )
-    parser.add_argument(
-        "-d",
-        "--delay",
-        type=int,
-        required=True,
-        help="Number of delay cycles to use as input to model.",
-    )
 
     FLAGS = parser.parse_args()
-    if (FLAGS.protocol != "http") and (FLAGS.protocol != "grpc"):
-        print(
-            'unexpected protocol "{}", expects "http" or "grpc"'.format(FLAGS.protocol)
-        )
-        exit(1)
-
-    client_util = httpclient if FLAGS.protocol == "http" else grpcclient
 
     # Run the busyop model which takes a delay as input.
     model_name = FLAGS.model
 
-    # Create the inference context for the model.
-    client = client_util.InferenceServerClient(FLAGS.url, verbose=FLAGS.verbose)
+    # Create the inference context for the model. Need to set the concurrency
+    # based on the number of requests so that the delivery of the async
+    # requests is not blocked.
+    # See the comment for more details: https://github.com/triton-inference-server/client/blob/r24.02/src/python/library/tritonclient/http/_client.py#L1501
+    client = httpclient.InferenceServerClient(
+        FLAGS.url, verbose=FLAGS.verbose, concurrency=FLAGS.num_requests
+    )
 
     # Collect async requests here
     requests = []
 
-    # Create the data for the one input tensor
-    input_data = np.array([FLAGS.delay], dtype=np.int32)
+    # Create the data for the input tensor. Creating tensor size with 5 MB.
+    tensor_size = [1, 5 * 1024 * 1024]
+    input_data = np.random.randn(*tensor_size).astype(np.float32)
 
     inputs = [
-        client_util.InferInput(
-            "in", input_data.shape, np_to_triton_dtype(input_data.dtype)
+        httpclient.InferInput(
+            "INPUT0", input_data.shape, np_to_triton_dtype(input_data.dtype)
         )
     ]
     inputs[0].set_data_from_numpy(input_data)
 
     # Send requests
     for i in range(FLAGS.num_requests):
         requests.append(client.async_infer(model_name, inputs))
-        print("Sent request %d" % i)
+        time.sleep(0.5)
```

Comment:
Is this sleep needed?

Reply:
I think we can still see the growth without the sleep. Removed this line.